app/uk/gov/hmrc/gform/sharedmodel/formtemplate/destinations/Destination.scala

```diff
@@ -411,4 +423,28 @@ object UploadableConditioning {
 
   def conditionAndValidate(convertSingleQuotes: Option[Boolean], os: Option[String]): Either[String, Option[String]] =
     os map (conditionAndValidate(convertSingleQuotes, _) map (_.some)) getOrElse Right(None)
+
+  def readFromURL(id: DestinationId, url: String) = {
+    if (!isRelativePath(url))
+      throw new Exception(
+        s"`payloadUrl` in the ${id.id} destination is not valid. The `payloadUrl` must be specified as a relative path."
+      )
+    val path = Paths.get(url).normalize().toAbsolutePath().toString()
+    val source = Source.fromFile(path)
+    try Some(source.mkString)
+    catch {
+      case ex: Exception =>
+        throw new RuntimeException(
+          s"An error occurs when attempting to read the file from $path in the ${id.id} destination with exception : $ex"
+        )
+    } finally source.close()
+  }
```

Comment:
What is the idea behind this? I'm not sure what is this for?

Reply:
The idea is to read and load the handlebar configuration into the `payload` attribute while posting a form template if payloadUrl is present.